app-misc/gcstar/gcstar-1.7.3.ebuild

```diff
@@ -0,0 +1,101 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# GUIDE:
+# - find dependencies:
+#   # ebuild gcstar-1.7.3.ebuild install
+# - find optional components, useful to determine IUSE flags:
+#   # grep checkModule /.../work/* -R
+#   # grep checkOptionalModule /.../work/* -R
+# - find all modules:
+#   # grep -E 'use .+;' /.../work/* -R
+
+# TODO:
+# - missing gtkspell:3 perl binding (dev-perl/gtk2-spell is masked)
+
+EAPI=7
+
+inherit desktop xdg-utils
+
+DESCRIPTION="Manage your collections of movies, games, books, music and more"
+HOMEPAGE="http://www.gcstar.org/"
+# Gna permanently shut down in May 2017
+# Original SRC_URI was http://download.gna.org/gcstar/${P}.tar.gz
+SRC_URI="https://gitlab.com/Kerenoc/GCstar/-/archive/v${PV}/GCstar-v${PV}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="cddb exif mp3 stats vorbis"
+#IUSE+="spell"
+
+#	spell? ( dev-perl/gtk2-spell )
+RDEPEND="dev-lang/perl[ithreads]
+	dev-perl/Gtk3
+	dev-perl/Gtk3-SimpleList
+	dev-perl/XML-Parser
+	dev-perl/XML-Simple
+	dev-perl/libwww-perl[ssl]
+	dev-perl/Archive-Zip
+	dev-perl/JSON
+	dev-perl/Locale-Codes
+	cddb? ( dev-perl/Net-FreeDB )
+	exif? ( media-libs/exiftool )
+	mp3? ( ||
+		(
+			dev-perl/MP3-Info
+			dev-perl/MP3-Tag
+		)
+	)
```

Comment:
This is markedly different to `1.7.1-r1`; was the old ebuild wrong in that not both of these packages are necessary for correct function of this package?

Reply:
Looking into the code, there's a condition to try `MP3::Info` first, and if it fails switch to `MP3::Tag`.

Should the ebuild pull them both anyway?

```
gcstar-1.7.1-r1: lib/gcstar/GCExtract/GCExtractMusics.pm

        ...

        $self->{hasMP3Info} = $self->checkModule('MP3::Info');
        $self->{hasMP3Tag} = $self->checkModule('MP3::Tag');

        ...

            if ($self->{hasMP3Info})
            {
                MP3::Info::use_mp3_utf8(1);
                my $song = MP3::Info::get_mp3tag($self->{firstTrack});
                $info->{title} = $song->{ALBUM};
                $info->{artist} = $song->{ARTIST};
                $info->{release} = $song->{YEAR};
                $info->{genre} = $song->{GENRE};
            }
            elsif ($self->{hasMP3Tag})
            {
                my $song = MP3::Tag->new($self->{firstTrack});
                (undef, undef, $info->{artist}, $info->{title}) = $song->autoinfo;
            }

        ...
```